### PR TITLE
Update linting workflow.

### PR DESCRIPTION
**Pull request by cqc-alec**

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -16,12 +16,12 @@ jobs:
       with:
         python-version: '3.x'
     - name: Install black and pylint
-      run: pip install black~=22.3 pylint~=2.13,!=2.13.6
+      run: pip install black>=24.4.2 pylint>=3.2.3
```
[the remark this answers is not in the document]
Comment (cqc-alec) [reply .github/workflows/lint.yml:19]: We did, but I'm inclined to think that the advantages (allowing developers to keep up-to-date versions of packages) outweigh the disadvantages (sometimes seeing CI runs fail unexpectedly, necessitating a new PR to fix). (These dependencies don't get picked up by dependabot.)

[the remark this answers is not in the document]
Comment (cqc-alec) [reply .github/workflows/lint.yml:19]: Perhaps another argument for doing this: I have my editor set up to automatically apply black formatting to python files on save. For this I need to have a single black version used across all projects. If each project is defining its own black version this gets more difficult to manage (though not impossible).

Comment (cqc-alec) [inline .github/workflows/lint.yml:19]:
```suggestion
      run: pip install black pylint
```
May as well just do this.

**Comment by cqc-melf** [comment]
Should we add a task to update the extensions as well?

**Review by cqc-melf — APPROVED** [review]

**Comment by cqc-alec** [comment]
> Should we add a task to update the extensions as well?

Yes, I can do that.

**Comment by cqc-alec** [comment]
> > Should we add a task to update the extensions as well?
> 
> Yes, I can do that.

https://github.com/CQCL-DEV/private-tket-issues/issues/33

**Pull request merged**